python/ml4ir/base/model/optimizers/optimizer.py

```diff
@@ -31,28 +44,8 @@ def choose_optimizer(model_config, learning_rate_schedule):
     else:
         optimizer_key = model_config['optimizer']['key']
         gradient_clip_value = model_config['optimizer']['gradient_clip_value']
-        if optimizer_key == OptimizerKey.ADAM:
-            return tf_optimizers.Adam(
-                learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value if 'gradient_clip_value' in model_config['optimizer'] else 5.0
-            )
-        elif optimizer_key == OptimizerKey.NADAM:
-            return tf_optimizers.Nadam(
-                learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value if 'gradient_clip_value' in model_config['optimizer'] else 5.0
-            )
-        elif optimizer_key == OptimizerKey.ADAGRAD:
-            return tf_optimizers.Adagrad(
-                learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value if 'gradient_clip_value' in model_config['optimizer'] else 5.0
-            )
-        elif optimizer_key == OptimizerKey.SGD:
-            return tf_optimizers.SGD(
-                learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value if 'gradient_clip_value' in model_config['optimizer'] else 5.0
-            )
-        elif optimizer_key == OptimizerKey.RMS_PROP:
-            return tf_optimizers.RMSprop(
-                learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value if 'gradient_clip_value' in model_config['optimizer'] else 5.0
-            )
-        else:
-            raise ValueError("Unsupported Optimizer: " + optimizer_key)
+        config = {'learning_rate':learning_rate_schedule, 'clipvalue': gradient_clip_value if 'gradient_clip_value' in model_config['optimizer'] else DefaultValues.GRADIENT_CLIP_VALUE}
+        return tf.keras.optimizers.get({'class_name':optimizer_key, 'config':config})
```

Comment:
Nice! I tried to do the same [in this PR](https://github.com/salesforce/ml4ir/pull/65); I like your solution too (i am closing the other). A shortcoming though is that there is manual care for two things (gradient_clip_val and learning_rate). Also, even if the user does not ask for `clipvalue` we apply it under the hood! We should re-think this in another PR.

Reply:
I agree. We can set the default gradient_clip value to be 'unbounded'.

Reply:
The reason I set a clipvalue by default is because I observed constant gradient explosion otherwise. We can remove the default if we all agree that we set it to some sensible value every time we train.

@mohazahran Thanks for this change. See! So clean now. :)